lib/active_model/serializer.rb

```diff
@@ -102,7 +111,16 @@ def self.adapter
     end
 
     def self.root_name
-      name.demodulize.underscore.sub(/_serializer$/, '') if name
+      name.demodulize.underscore.sub(/_serialization$/, '') if name
+    end
+
+    def self.deserialize(params)
+      association_keys = _reflections.map(&:name)
+      permitted_params = adapter.params(@_params, association_keys) || @_params
+      permitted_key    = adapter.root || root_name
+
+      sanitazed_params = params.require(permitted_key.to_sym).permit(permitted_params)
+      adapter.parse(sanitazed_params)
```

Comment:
@joaomdmoura As mentioned [there](https://github.com/joaomdmoura/active_model_serializers/pull/6) I forgot to remove this method while merging our forks (you can see that currently the method `deserialize` is defined twice).